pkg/policies/autoscale/kubernetes/discovery/autoscale-control-points.go

```diff
@@ -96,13 +101,32 @@ var _ AutoScaleControlPointStore = (*autoScaleControlPoints)(nil)
 var _ AutoScaleControlPoints = (*autoScaleControlPoints)(nil)
 
 // newAutoScaleControlPoints returns a new AutoScaleControlPoints.
-func newAutoScaleControlPoints(trackers notifiers.Trackers, k8sClient k8s.K8sClient) *autoScaleControlPoints {
-	return &autoScaleControlPoints{
+func newAutoScaleControlPoints(trackers notifiers.Trackers, k8sClient k8s.K8sClient, pr *prometheus.Registry, election *election.Election) (*autoScaleControlPoints, error) {
+	cpc := &autoScaleControlPoints{
 		controlPoints: make(map[AutoScaleControlPoint]*controlPointState),
 		trackers:      trackers,
 		k8sClient:     k8sClient,
 		scaleStream:   stream.New(),
 	}
+	if election != nil && election.IsLeader() {
```

Comment:
This is a race condition. This code will probably hit before leader election. So, either check for election results periodically like [here](https://github.com/fluxninja/aperture/blob/main/pkg/policies/autoscale/kubernetes/discovery/autoscale-discovery.go#L60) or subscribe to results using notifier tracker as [here](https://github.com/fluxninja/aperture/blob/main/pkg/policies/autoscale/kubernetes/actuators/podscaler/pod-scaler.go#L57)